test/inductor/test_torchinductor.py

```diff
@@ -386,9 +390,16 @@ def run(*ex, **kwargs):
             self.assertEqual(len(results_that_require_grad), 0)
         else:
             actual_grad = compute_grads(example_inputs, kwargs, actual, grads)
+            flat_actual, _ = tree_flatten(actual_grad)
+
+            if reference_in_float:
+                flat_expect = reference_to_expect(flat_actual, flat_grads)
+            else:
+                flat_expect = flat_grads
+
             self.assertEqual(
-                actual_grad,
-                correct_grad,
+                flat_actual,
+                flat_expect,
```

Comment:
In the forward they unflatten it before passing it to assertEqual, perhasp you should do here the same?